api/naming_service.go

```diff
@@ -122,3 +123,21 @@ func handleLookup(name string, coins []uint64) (result []blockatlas.Resolved, er
 	result, err = api.Lookup(coins, name)
 	return
 }
+
+// Obtain tld from then name, e.g. ".ens" from "nick.ens"
+func getTLD(name string) (tld string, ok bool) {
+	if strings.Index(name, ".") >= 0 {
+		// xxx.tld format
+		ss := strings.Split(name, ".")
+		tld := "." + ss[len(ss)-1]
+		return tld, true
+	}
+	if strings.Index(name, "@") >= 0 {
+		// xxx@tld format
```

Comment:
Is it possbile to have something like: xxx.ens@fio?

Reply:
TLD separation logic reworked, this special case is handled now, unit test added.